.github/workflows/deploy.yml

```diff
@@ -0,0 +1,125 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# Deploy a new version of the documentation or rebuild an existing one.
+
+# Note: All changes made to the gh-pages branch are non-destructive 
+#       (i.e. no force pushing) so all changes can be undone.
+
+name: deploy
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag present on both cylc-flow and cylc-docs.'
+        required: true
+      set_current:
+        description: 'Make this the default documented version.'
+        required: false
+        default: true
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: configure python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.7'
+
+      - name: checkout cylc-doc
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.inputs.tag }}
+          path: docs
+
+      - name: install graphviz
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y graphviz pkg-config libgraphviz-dev
+          pip install pygraphviz
+
+      - name: install cylc-doc
+        run: |
+          pip install "${{ github.workspace }}/docs[all]"
+
+      - name: install cylc-flow
+        run: |
+          # NOTE: Install with [all] so we can import plugins which may
+          #       have extra dependencies.
+          pip install "cylc-flow[all]==${{ github.event.inputs.tag }}"
+
+      - name: checkout gh-pages
+        uses: actions/checkout@v2
+        with:
+          ref: gh-pages
+          path: gh-pages
+
+      - name: sync static files
+        run: |
+          DOCS="${{ github.workspace }}/docs" \
+          PAGE="${{ github.workspace }}/gh-pages"  \
+
+          rsync -r "$DOCS/doc/" "$PAGE/"
+
+          ls "$PAGE"
+
+      - name: install gh-pages
+        run: |
+          DOCS="${{ github.workspace }}/docs" \
+          PAGE="${{ github.workspace }}/gh-pages"  \
+
+          (
+            cd "$DOCS"
+            rm -r doc
+            ln -s "$PAGE" doc
+          )
+
+          ls -l "$DOCS"
+
+      - name: build docs
+        run: |
+          # NOTE: Doing a clean build allows up to update docs for a
+          #       deployed version at a later date.
+          make -C docs \
+            clean \
+            html \
+            slides \
+            linkcheck \
+            SPHINXOPTS='-Wn' \
+            SETCURRENT=${{ github.event.inputs.set_current }}
+
+      - name: configure git
+        run: |
+          git config --global user.name "action:deploy"
+          git config --global user.email "action:deploy@github.com"
+
+      - name: tidy old versions
+        run: |
+          (
+            cd docs
+            git rm -r $("${{ github.workspace }}/docs/bin/version" tidy) || true
```

Comment:
[From the demo run:](https://github.com/oliver-sanders/cylc-doc/runs/878320584?check_suite_focus=true)
> ```
> line 3: /home/runner/work/cylc-doc/cylc-doc/docs/bin/version: No such file or directory
> fatal: No pathspec was given. Which files should I remove?
> ```
Is this expected?

Reply:
Yes, ugly I know but it's the natural result of `git rm -r  # nothing`.